include/gamepad/button.hpp

```diff
@@ -84,7 +84,7 @@ class Button {
          *   gamepad::master.Up.onPress("upPress1", []() { std::cout << "I was pressed!" << std::endl; });
          * @endcode
          */
-        bool onPress(std::string listenerName, std::function<void(void)> func) const;
+        uint32_t onPress(std::string listenerName, std::function<void(void)> func) const;
```

Comment:
the functions return `INT_MAX` when errno is set, which is the maximum value of an `int32_t`. Consider changing them to `int32_t` for clarity, but this is not a requirement

```suggestion
        int32_t onPress(std::string listenerName, std::function<void(void)> func) const;
```